### ISSUE TITLE
Elasticsearch no longer runs on MacOS Ventura due to `jdk` being damaged

### ISSUE THREAD

**Issue by tomtaylor** [elastic/homebrew-tap#141]
Installing `elastic/tap/elasticsearch-full` works, but it doesn't run as a background service. MacOS presents an error messages saying that the `jdk` executable is damaged and prompts the user to remove it. Running `elasticsearch` in a shell works if you have a local Java install defined using the `ES_JAVA_HOME` environment variable. I suspect this means the embedded JDK is somehow incompatible with MacOS Ventura and needs updating.

While I understand this tap is now deprecated, we use it because we don't love the overhead of Docker. It would be great to get an update to resolve this, or to remove this tap entirely so people don't get confused trying to use it.

**Comment by Hjsurk** [comment]
@mgreau Sorry for the direct tag, but any chance we could get a possible update on this?
This repo is still used a lot by MacOS users and now suddenly not functional for MacOS Ventura.

**Comment by dougal** [comment]
A workaround based on what @tomtaylor said above:

Install another Java. I installed OpenJDK.

```shell
brew install openjdk
```

I then added the `ES_JAVA_HOME` environment variable to the `plist` file used to launch elasticsearch by `brew services`. This environment variable was pointed to the OpenJDK directory.

Note I am on an Apple Silicon Mac, so `brew --prefix` is `/opt/homebrew`.

Edit the file `/opt/homebrew/Cellar/elasticsearch-full/7.17.4/homebrew.mxcl.elasticsearch-full.plist`, changing the environment variables dictionary to look like:

```
    <key>EnvironmentVariables</key>
    <dict>
      <key>ES_JAVA_HOME</key>
      <string>/opt/homebrew/opt/openjdk</string>
    </dict>
```

`brew services start elasticsearch-full` was then successful.

**Comment by Vashiru** [comment]
That did the trick for me as well. For future reference that's OpenJDK `19.0.1` and Elasticsearch-full `7.17.4`.

Yes there's also `~/Library/LaunchAgents/homebrew.mxcl.elasticsearch-full.plist`, no that's not the one you edit. Edit the one in `/opt/homebrew/Cellar/elasticsearch-full/7.17.4/homebrew.mxcl.elasticsearch-full.plist`. As mentioned, `/opt/homebrew` is the prefix for M1 macs. For intel based Macs this is `/usr/local/` so that would make the path most likely: `/usr/local/Cellar/elasticsearch-full/7.17.4/homebrew.mxcl.elasticsearch-full.plist`. Unsure? You can always run `brew --prefix` to find out where to look.

**Comment by nkarthickannan** [comment]
After following the steps provided by @dougal I had to reload the plist file to get work in my Mac:
```
launchctl unload /opt/homebrew/Cellar/elasticsearch-full/7.17.4/homebrew.mxcl.elasticsearch-full.plist
launchctl load /opt/homebrew/Cellar/elasticsearch-full/7.17.4/homebrew.mxcl.elasticsearch-full.plist
```

**Comment by drnic** [comment]
Thanks for the above.

Thoughts on telling the CLI where openjdk is?

```
$ elasticsearch --version
could not find java in bundled JDK at /opt/homebrew/Cellar/elasticsearch-full/7.17.4/libexec/jdk.app/Contents/Home/bin/java
```

**Comment by drnic** [comment]
I'm on Apple Silicon and was getting the error "Failure running machine learning native code. This could be due to running on an unsupported OS or distribution, missing OS libraries, or a problem with the temp directory. To bypass this problem by running Elasticsearch without machine learning functionality set [xpack.ml.enabled: false]."

To make this go away I edited the config file: `$(brew --prefix elasticsearch-full)/libexec/config/elasticsearch.yml` and added the following to the bottom:

```yaml
xpack.ml.enabled: false
```

And restarted:

```
brew services restart elastic/tap/elasticsearch-full
```

**Comment by redaxle74** [comment]
thanks @dougal :)

**Comment by kristian-nst** [comment]
2025 anyone? 
```
tail elasticsearch.log
/opt/homebrew/Cellar/elasticsearch-full/7.17.4/libexec/bin/elasticsearch-env: line 83: 99796 Killed: 9               "$JAVA" "$XSHARE" -cp "$ES_CLASSPATH" org.elasticsearch.tools.java_version_checker.JavaVersionChecker
Error occurred during initialization of VM
java.lang.Error: A command line option has attempted to allow or enable the Security Manager. Enabling a Security Manager is not supported.
	at java.lang.System.initPhase3(java.base@24.0.2/System.java:1947)

Error occurred during initialization of VM
java.lang.Error: A command line option has attempted to allow or enable the Security Manager. Enabling a Security Manager is not supported.
	at java.lang.System.initPhase3(java.base@24.0.2/System.java:1947)
```

**Comment by natematykiewicz** [comment]
@kristian-nst did you ever solve this?

**Comment by kristian-nst** [comment]
> [@kristian-nst](https://github.com/kristian-nst) did you ever solve this?

Nope, got fed up and switched to meilisearch 🙄 (way less memory usage, at least in my deployment)

**Comment by gokerDEV** [comment]
`brew install openjdk@21`

Export  fullpath
```
export ES_JAVA_HOME="/opt/homebrew/opt/openjdk@21/libexec/openjdk.jdk/Contents/Home"
```
Ta da 
```
elasticsearch --version
Version: 7.17.4, Build: default/tar/79878662c54c886ae89206c685d9f1051a9d6411/2022-05-18T18:04:20.964345128Z, JVM: 21.0.9
```
